homeassistant/components/openweathermap/weather.py

```diff
@@ -60,8 +60,9 @@ def unique_id(self):
     @property
     def device_info(self):
         """Return the device info."""
+        split_unique_id = self._unique_id.split("-")
         return {
-            "identifiers": {(DOMAIN, self._unique_id)},
+            "identifiers": {(DOMAIN, f"{split_unique_id[0]}-{split_unique_id[1]}")},
```

Comment:
If you change the unique ID, you'll need to migrate existing entries/devices so it is not breaking.

Reply:
This specifically keep the existing device_id per location on purpose .
Do I miss your point?

Reply:
> Do I miss your point?

Yes, this basically creates a new device. Instead, existing devices in the device registry need to be updated/migrated to have the new ID.

Reply:
> Yes, this basically creates a new device

Unless I'm blinded, it doesn't.
Currently, the device_id = unique_id = lat-lon
The new unique_id is lat-lon-mode, so this extract lat-lon from the unique_id to keep the existing device_id

Reply:
Right... see review comment above 😉